src/command_line_parser.cc

```diff
@@ -1235,6 +1235,11 @@ TritonParser::Parse(int argc, char** argv)
   triton::server::grpc::Options& lgrpc_options = lparams.grpc_options_;
 #endif  // TRITON_ENABLE_GRPC
 
+#if defined TRITON_ENABLE_HTTP || defined TRITON_ENABLE_GRPC
+  // According to HTTP specification header names are case-insensitive.
+  const std::string case_insensitive_prefix{"(?i)"};
```

Comment:
Looks clean 🚀 I think this approach will also let users disable the case insensitivity if they **really** need to for some unforeseen reason. ex: `--header-pattern "(?-i)..."` to disable it, which I think is a good flexibility to have.